### DISCUSSION TITLE
Gtk.MessageDialog.New can only use 0 arguments

### DISCUSSION THREAD

**Discussion by PlatinumLucario** [gircore/gir.core#877] [Q&A] [answered]
[Unlike the constructor mentioned in the official docs](https://docs.gtk.org/gtk4/ctor.MessageDialog.new.html), it appears that Gtk.MessageDialog.New() can only use 0 arguments. The official docs state that ``gtk_message_dialog_new()`` contains 5 arguments.

Furthermore, it does state that "This method is not directly available to language bindings." in the official docs. Could this be why it can only show as 0 arguments when attempting to use the ``New()`` method in MessageDialog?

I am aware there's a ``Adw.MessageDialog.New()``, which can use 3 arguments, which the [binding directly lines up with it's official documentation](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1-latest/class.MessageDialog.html).

So I'm really curious, was ``Gtk.MessageDialog.New()`` arguments meant to be unavailable to any language bindings?

**Comment by cameronwhite** [comment]
I think the issue with that method is that `gtk_message_dialog_new()` allows a variadic number of arguments (note the `...` at the end) due to the string formatting support. Variadic methods would be more complex to wrap from other language bindings and I'm pretty sure gir.core doesn't support them, but the method may also be omitted from the gir file in the first place based on that documentation comment

The workaround would be to use the default constructor and then set the necessary properties afterwards

**Reply by badcel** [reply, marked as answer]
Yes variadic parameters are not supported currently, so the constructor is missing.

The documentation comment regarding "not available to language bindings" states that this function is not part of the typelib file. As gir.core does not use the typelib file the comment does not apply.
